application-ckeditor-plugins/src/main/resources/xwiki-image/plugin.js

```diff
@@ -17,224 +17,117 @@
+      }).done(function(data) {
+        if (widget && widget.element) {
+          widget.setData(data);
+        } else {
+          var element = CKEDITOR.dom.element.createFromHtml(widget.template.output(), editor.document);
+          var wrapper = editor.widgets.wrapElement(element, widget.name);
+          var temp = new CKEDITOR.dom.documentFragment(wrapper.getDocument());
+
+          // Append wrapper to a temporary document. This will unify the environment in which #data listeners work when
+          // creating and editing widget.
+          temp.append(wrapper);
+
+          var instance = editor.widgets.initOn(element, widget, {});
+          instance.edit();
+          editor.widgets.initOn(element, widget, data);
+          editor.widgets.finalizeCreation(temp);
         }
       });
+    });
+  }
 
-      editor.balloonToolbars.create({
-        buttons: 'Link,Unlink,Image',
-        widgets: 'image'
-      });
+  CKEDITOR.plugins.add('xwiki-image', {
+    requires: 'xwiki-image-old', // TODO: maybe require modal to have skinx?
+    init: function(editor) {
+      this.initImageDialogWidget(editor);
     },
+    initImageDialogWidget: function(editor) {
+      var imagePlugin = this;
+      var imageWidget = editor.widgets.registered.image;
+      this.overrideImageWidget(editor, imageWidget);
 
+      imageWidget.insert = function() {
+        showImageWizard(editor, this);
+      };
+      imageWidget.edit = function(event) {
+        // Prevent the default behavior because we want to use our custom image dialog.
+        event.cancel();
+        showImageWizard(editor, this);
+      };
+    },
     overrideImageWidget: function(editor, imageWidget) {
+      CKEDITOR.plugins.registered['xwiki-image-old'].overrideImageWidget(editor, imageWidget);
+
       var originalInit = imageWidget.init;
       imageWidget.init = function() {
         originalInit.call(this);
-        var serializedResourceReference = this.parts.image.getAttribute('data-reference');
-        if (serializedResourceReference) {
-          this.setData('resourceReference', CKEDITOR.plugins.xwikiResource
-            .parseResourceReference(serializedResourceReference));
+
+        // Caption
+        if (this.parts.caption) {
+          this.setData('hasCaption', true);
+          // TODO: Add support for editing the caption directly from the dialog (see CKEDITOR-435)
+          //this.setData('imageCaption', this.parts.caption.getText())
+        } else {
+          this.setData('hasCaption', false);
         }
+
+        // Style
+        this.setData('imageStyle', this.parts.image.getAttribute('data-ckeditor-image-style'));
+
+        this.setData('border', this.parts.image.getAttribute('data-ckeditor-image-style-border'));
+        this.setData('alignment', this.parts.image.getAttribute('data-ckeditor-image-style-alignment') || 'none');
+        this.setData('textWrap', this.parts.image.getAttribute('data-ckeditor-image-style-textWrap'));
```

Comment:
This part needs to be discussed, we have two options:
- using `data-*` attribute as it is done currently
- using a set of predetermined classes